### PR TITLE
Remove fuzzy set from experimental

### PR DESCRIPTION
**Pull request by mgodwan**
### Description
https://github.com/opensearch-project/OpenSearch/issues/12126

### Related Issues
https://github.com/opensearch-project/OpenSearch/issues/12126

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md
Comment (shwetathareja) [inline CHANGELOG.md:139]: Mark fuzzy filter GA and remove experimental setting

**Review by shwetathareja — APPROVED** [review]

**Comment by shwetathareja** [comment]
Unrelated test failures

[org.opensearch.telemetry.metrics.TelemetryMetricsEnabledSanityIT.testObservableGauge](https://build.ci.opensearch.org/job/gradle-check/35125/testReport/junit/org.opensearch.telemetry.metrics/TelemetryMetricsEnabledSanityIT/testObservableGauge/)
    [org.opensearch.index.shard.IndexShardTests.testCommitLevelRestoreShardFromRemoteStore](https://build.ci.opensearch.org/job/gradle-check/35125/testReport/junit/org.opensearch.index.shard/IndexShardTests/testCommitLevelRestoreShardFromRemoteStore/)

**Comment by reta** [comment]
@shwetathareja why we are merging pull request with failing Gradle checks?

**Comment by shwetathareja** [comment]
> @shwetathareja why we are merging pull request with failing Gradle checks?

@reta those were known test flaky test failures. I have added the comment as well before merging.

**Comment by reta** [comment]
@shwetathareja please do not merge pull requests when checks are red, the flaky test procedure is described in description, but it is difficult to understand when test becomes non flaky because of change in the request itself, green checks at least give some guarantees, I hope you would agree. Thank you.

**Comment by shwetathareja** [comment]
@reta : Is there a consensus from all maintainers to not merge with gradle check failing due to known flaky test failures (with same stack traces)? Gradle checks don't pass even after multiple attempts. 

>  it is difficult to understand when test becomes non flaky because of change in the request itself,

This may not be true with green gradle check after 10 or so failed attempts.

Also, authors don't have even permission to trigger gradle check re-run.

**Comment by mgodwan** [comment]
> Also, authors don't have even permission to trigger gradle check re-run.

+1, it is not feasible to trigger a gradle check run only. As authors, we usually need to put an empty commit just to trigger gradle checks which does not seem correct.

**Comment by shwetathareja** [comment]
> we have checks for a reason right?

Those reasons shouldn't be wasting Authors and maintainers time, re-running gradle-check for the whole day.

**Comment by reta** [comment]
> Those reasons shouldn't be wasting Authors and maintainers time, re-running gradle-check for the whole day.

Totally, so why we (maintainers) led to that to happen? Why instead of merging with failed check not to go mute / fix / remove test? You are a maintainer as well - please help.

**Comment by dblock** [comment]
My 0.02c is that if we start merging changes with "unrelated failed tests", we will never have a green build. I am with @reta to keep the requirement of a passing gradle check, unrelated failure or not.

**Comment by reta** [comment]
@shwetathareja we had a bunch of initiative to address that (we very low participation from maintainers), just to name a few that were formalized:
 - https://github.com/opensearch-project/OpenSearch/issues/10371
 - https://github.com/opensearch-project/OpenSearch/issues/9462

**Pull request merged**
